File: src/algorithms/algo83/algo83.py
class BinaryTree:
    class Node:
        def __init__(self, value):
            self.value = value
            self.left = None
            self.right = None

    def __init__(self):
        self.root = None

    def invert(self):
        self._invert(self.root)

    def _invert(self, node):
        if node is None:
            return

        node.left, node.right = node.right, node.left

        self._invert(node.left)
        self._invert(node.right)

File: src/algorithms/algo83/Algo83.java
package algorithms.algo83;

public class Algo83 {
    

    private Node root;

    private class Node {
        private int value;
        private Node left;
        private Node right;

        public Node(int value) {
            this.value = value;
            this.left = null;
            this.right = null;
        }
    }

    public void invert() {
        invert(root);
    }

    private void invert(Node node) {
        if (node == null) {
            return;
        }

        Node temp = node.left;
        node.left = node.right;
        node.right = temp;

        invert(node.left);
        invert(node.right);
    }
}
